### DISCUSSION TITLE
One Model, One Resource how to change the resource Eloquent query

### DISCUSSION THREAD

**Discussion by cirrion-software** [filamentphp/filament#11072] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.14

### How can we help you?

I have a SalesDocument model which is used for both Quotations and Orders. One table in the database sales_documents. I have set up a resource SalesDocumentResource which obviously when listing records shows a mixture of Quotations and Orders. There is a field `_is_order_` which determines if it is a quotation or order. What I would like is 2 menu items in the sidebar, one for Orders, the other of Quotations which links to the resource table but will update the Eloquent query in the resource depending on which link is followed. I know I could use tabs to split these up on the resource table but I need to use tabs to split up the Quotations and Orders by status so cannot use them.

Any help would be appreciated, pulling my hair out.

I know the resource query can be edited with [https://filamentphp.com/docs/3.x/panels/resources/getting-started#customizing-the-resource-eloquent-query](url) but how do change this depending on the link clicked to the resource and how do I get the links in the sidebar?

**Comment by leandrocfe** [comment, marked as answer]
You can create QuotationResource and OrderResource, both using the SalesDocument model in the `$model` attribute
```php
protected static ?string $model = SalesDocument::class;
```
You can modify the query in the table

```php
$table
    ->modifyQueryUsing(function (Builder $query) {
        $query->where('is_order', true);// or false
    })
```
